server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/utils/StarTreeMetaWriter.java

```diff
@@ -0,0 +1,226 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.compositeindex.datacube.startree.utils;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.index.SegmentWriteState;
+import org.apache.lucene.store.IndexOutput;
+import org.opensearch.index.compositeindex.datacube.Dimension;
+import org.opensearch.index.compositeindex.datacube.startree.StarTreeField;
+import org.opensearch.index.compositeindex.datacube.startree.aggregators.MetricAggregatorInfo;
+import org.opensearch.index.mapper.CompositeMappedFieldType;
+
+import java.io.IOException;
+import java.util.List;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.opensearch.index.compositeindex.CompositeIndexConstants.MAGIC_MARKER;
+import static org.opensearch.index.compositeindex.CompositeIndexConstants.VERSION;
+
+/**
+ * The utility class for serializing the metadata of a star-tree data structure.
+ * The metadata includes information about the dimensions, metrics, and other relevant details
+ * related to the star tree.
+ *
+ * @opensearch.experimental
+ */
+public class StarTreeMetaWriter {
+
+    private static final Logger logger = LogManager.getLogger(StarTreeMetaWriter.class);
+
+    /**
+     * Serializes the star-tree metadata.
+     *
+     * @param metaOut                the IndexOutput to write the metadata
+     * @param compositeFieldType     the composite field type of the star-tree field
+     * @param starTreeField          the star-tree field
+     * @param writeState             the segment write state
+     * @param metricAggregatorInfos  the list of metric aggregator information
+     * @param segmentAggregatedCount the aggregated document count for the segment
+     * @param dataFilePointer        the file pointer to the start of the star tree data
+     * @param dataFileLength         the length of the star tree data file
+     * @throws IOException if an I/O error occurs while serializing the metadata
+     */
+    public static void serializeStarTreeMetadata(
+        IndexOutput metaOut,
+        CompositeMappedFieldType.CompositeFieldType compositeFieldType,
```

Comment:
Since you're already in StarTreeMetaWriter, do you need the type as part of the constructor ?